lib/srv/sess.go

```diff
@@ -282,7 +282,7 @@ func (s *SessionRegistry) TryCreateHostUser(identityContext IdentityContext) (cr
 	if err != nil {
 		if trace.IsAccessDenied(err) {
 			log.Warnf("Unable to create host users: %v", err)
-			return false, nil, nil
+			return false, nil, trace.Wrap(err)
```

Comment:
Unfortunately you can't just return the error here,`TryCreateHostUser` is called for every incoming SSH connection, if the user isn't allowed to create host users but the user already exists, that's fine and we should allow the connection. I'm pretty sure that's why some of the tests are failing on this PR.

We only really want to return the error if the host user doesn't already exist. I think the existing code "tries" to return this error below with "Insufficient permission for host user creation", but there is a logic bug, `trace.IsAccessDenied(err)` will always be false or else the function would have already returned

Reply:
Great call out :+1: I should've noticed we were specifically allowing connections a couple of lines down  🤦 I think these checks are _slightly_ less confusing now, and I also fixed up an existing case where a failed check against the access checker would result in an error being returned regardless of whether or not the host user already exists